src/corefx/System.Globalization.Native/locale.cpp

```diff
@@ -0,0 +1,373 @@
+			break;
+		case FractionalDigitsCount:
+			*value = 2; //TODO: implement
+			break;
+		case NegativeNumberFormat:
+			*value = 1; //TODO: implement
+			break;
+		case MonetaryFractionalDigitsCount:
+			*value = 2; //TODO: implement
+			break;
+		case PositiveMonetaryNumberFormat:
+			*value = 0; //TODO: implement
+			break;
+		case NegativeMonetaryNumberFormat:
+			*value = 0; //TODO: implement
+			break;
+		case CalendarType:
+			*value = 1; //TODO: implement
+			break;
+		case FirstWeekOfYear:
+			*value = 0; //TODO: implement
+			break;
+		case ReadingLayout:
+			*value = 0; //TODO: implement
+			break;
+		case NegativePercentFormat:
+			*value = 0; //TODO: implement
+			break;
+		case PositivePercentFormat:
+			*value = 0; //TODO: implement
+			break;
+		default:
+			status = U_UNSUPPORTED_ERROR;
+			break;
+	}
+
+	return status;
+}
+
+/*
+Function:
+GetLocaleInfoDecimalFormatSymbol
+
+Obtains the value of a DecimalFormatSymbols
+*/
+UErrorCode GetLocaleInfoDecimalFormatSymbol(const Locale &locale, DecimalFormatSymbols::ENumberFormatSymbol symbol, UChar* value, int32_t valueLength)
+{
+	UErrorCode status = U_ZERO_ERROR;
+	DecimalFormatSymbols *decimalsymbols = new DecimalFormatSymbols(locale, status);
+	if (decimalsymbols == NULL)
+	{
+		return U_MEMORY_ALLOCATION_ERROR;
+	}
+
+	if (U_FAILURE(status))
+	{
+		delete decimalsymbols;
+		return status;
+	}
+
+	UnicodeString s = decimalsymbols->getSymbol(symbol);
+	delete decimalsymbols;
+
+	s.extract(value, valueLength, status);
+	return status;
+}
+
+/*
+Function:
+GetDigitSymbol
+
+Obtains the value of a Digit DecimalFormatSymbols
+*/
+UErrorCode GetDigitSymbol(const Locale &locale, UErrorCode previousStatus, DecimalFormatSymbols::ENumberFormatSymbol symbol, int digit, UChar* value, int32_t valueLength)
+{
+	if (U_FAILURE(previousStatus))
+	{
+		return previousStatus;
+	}
+
+	return GetLocaleInfoDecimalFormatSymbol(locale, symbol, value + digit, valueLength - digit);
+}
+
+/*
+Function:
+GetLocaleInfoAmPm
+
+Obtains the value of a DateFormatSymbols Am or Pm string
+*/
+UErrorCode GetLocaleInfoAmPm(const Locale &locale, bool am, UChar* value, int32_t valueLength)
+{
+	UErrorCode status = U_ZERO_ERROR;
+	DateFormatSymbols *dateFormatSymbols = new DateFormatSymbols(locale, status);
+	if (dateFormatSymbols == NULL)
+	{
+		return U_MEMORY_ALLOCATION_ERROR;
+	}
+
+	if (U_FAILURE(status))
+	{
+		delete dateFormatSymbols;
+		return status;
+	}
+
+	int32_t count = 0;
+	const UnicodeString *tempStr = dateFormatSymbols->getAmPmStrings(count);
+	int offset = am ? 0 : 1;
+	if (offset >= count)
+	{
+		return U_INTERNAL_PROGRAM_ERROR;
+	}
+
+	tempStr[offset].extract(value, valueLength, status);
+	delete dateFormatSymbols;
+	return status;
+}
+
+/*
+PAL Function:
+GetLocaleInfoString
+
+Obtains string locale information
+*/
+extern "C" int32_t GetLocaleInfoString(const UChar* localeName, LocaleStringData localeStringData, UChar* value, int32_t valueLength)
```

Comment:
Is this used somewhere yet?  I expect we need to have a fixed return value at the boundary when valueLength is too small (instead of bubbling up U_BUFFER_OVERFLOW_ERROR.